src/Storages/Distributed/DistributedBlockOutputStream.h

```diff
@@ -84,7 +84,7 @@ class DistributedBlockOutputStream : public IBlockOutputStream
     std::string getCurrentStateDescription();
 
 private:
-    const Context & context;
+    std::shared_ptr<Context> context;
```

Comment:
Why is this change needed?

Reply:
Because context will be changed.

Reply:
I mean, why the shared ownership, maybe `unique_ptr` is enough? Or even copying `ClientInfo` instead of the entire context, because it is passed to RemoteBlockInputStream separately.

Reply:
Yes, `unique_ptr` is enough.